aptos-move/aptos-vm/src/aptos_vm_impl.rs

```diff
@@ -513,9 +516,12 @@ impl AptosVMImpl {
         txn_data: &TransactionMetadata,
         transaction_validation: &TransactionValidation,
     ) -> VMResult<()> {
-        let txn_sequence_number = txn_data.sequence_number();
+        let mut txn_sequence_number = txn_data.sequence_number();
         let txn_gas_price = txn_data.gas_unit_price();
         let txn_max_gas_units = txn_data.max_gas_amount();
+        if txn_data.with_gas_payer {
+            txn_sequence_number |= GAS_PAYER_FLAG_BIT;
+        }
```

Comment:
should this still be there?

Reply:
It's used to communicate with aptos-framework